lib/skate/detours/snapshot_serde.ex

```diff
@@ -27,4 +31,280 @@ defmodule Skate.Detours.SnapshotSerde do
   """
   def id_from_snapshot(%{"context" => %{"uuid" => id}}), do: id
   def id_from_snapshot(%{"context" => %{}}), do: nil
+
+  @doc """
+  Builds XState Snapshot from Detours Database object
+  """
+  def serialize(%Detour{} = detour) do
+    validate_serialized_snapshot(
+      %{
+        "value" => state_from_detour(detour),
+        "status" => "active",
+        "context" => context_from_detour(detour),
+        "children" => snapshot_children_from_detour(detour),
+        "historyValue" => %{}
+      },
+      detour
+    )
+  end
+
+  defp validate_serialized_snapshot(
+         serialized_snapshot,
+         %Detour{id: id, state: state}
+       ) do
+    if serialized_snapshot === state do
+      serialized_snapshot
+    else
+      log_fallback(id)
+      Logger.info(MapDiff.diff(state, serialized_snapshot))
+      state
+    end
+  end
+
+  # For each of these retrieve functions, the first function is the one
+  # that will look for the correct field on the db object, once the detour state
+  # is properly distributed across db tables / columns.
+  #
+  # The second function is the fallback that uses the existing state snapshot.
+  #
+  # Note that the first function is only stubbed out and will be significantly updated
+  # when the new fields are added to the db. For example, `get_route_from_db_detour`
+  # suggests that the field `route` will be surfaced at the top level of the detour
+  # but actually, it may be nested under `route_pattern` or however else we organize
+  # it in the db.
+
+  defp context_from_detour(%Detour{} = detour) do
+    %{
+      "uuid" => uuid_from_detour(detour),
+      "route" => route_from_detour(detour),
+      "routePattern" => routepattern_from_detour(detour),
+      "routePatterns" => routepatterns_from_detour(detour),
+      "startPoint" => startpoint_from_detour(detour),
+      "endPoint" => endpoint_from_detour(detour),
+      "waypoints" => waypoints_from_detour(detour),
+      "nearestIntersection" => nearestintersection_from_detour(detour),
+      "detourShape" => detourshape_from_detour(detour),
+      "finishedDetour" => finisheddetour_from_detour(detour),
+      "editedDirections" => editeddirections_from_detour(detour),
+      "selectedDuration" => selectedduration_from_detour(detour),
+      "selectedReason" => selectedreason_from_detour(detour)
+    }
+    |> Enum.filter(fn {_, v} -> v != nil end)
+    |> Enum.into(%{})
+  end
+
+  # defp state_from_detour(%Detour{detour_state: state}), do: state
+  defp state_from_detour(%Detour{
+         state: %{
+           "value" => state
+         }
+       }) do
+    log_fallback("state")
+    state
+  end
+
+  defp state_from_detour(_), do: nil
+
+  # defp uuid_from_detour(%Detour{detour_state: state}), do: state
+  defp uuid_from_detour(%Detour{
+         state: %{
+           "context" => %{
+             "uuid" => uuid
+           }
+         }
+       }) do
+    log_fallback("uuid")
+    uuid
+  end
```

Comment:
issue: I don't think we should fallback to the snapshot here, because the `id` is the authoritative field.

What do you think?

```suggestion
  defp uuid_from_detour(%Detour{id: id}) do
     id
  end
```

Reply:
or at the very least we should prioritize the `id` field.

Reply:
That's probably right! So, no fallback function here at all; the function will just fail